commit b00dc238aa393a6047b23df39269be00d2342ae2
Author: Raul Gutierrez Segales <rgs@collabora.co.uk>
Date:   Thu Sep 8 18:19:35 2011 +0100

    IndividualAggregator: copy the list of stores before looping over it
    
    Because a Backend's list of PersonaStores can change while we are
    iterating it, we do a copy before starting.
    
    Fixes: https://bugzilla.gnome.org/show_bug.cgi?id=658580

diff --git a/folks/individual-aggregator.vala b/folks/individual-aggregator.vala
index 4b158ae..a863800 100644
--- a/folks/individual-aggregator.vala
+++ b/folks/individual-aggregator.vala
@@ -548,8 +548,13 @@ public class Folks.IndividualAggregator : Object
           backend.notify["is-quiescent"].connect (
               this._backend_is_quiescent_changed_cb);
 
-          /* handle the stores that have already been signaled */
-          foreach (var persona_store in backend.persona_stores.values)
+          /* Handle the stores that have already been signaled. Since
+           * this might change while we are looping, get a copy first.
+           *
+           * https://bugzilla.gnome.org/show_bug.cgi?id=658580
+           */
+          var stores = backend.persona_stores.values.to_array ();
+          foreach (var persona_store in stores)
               {
                 this._backend_persona_store_added_cb (backend, persona_store);
               }
